python/ray/tune/tests/test_integration_wandb.py

```diff
@@ -92,337 +104,326 @@ class WandbTestTrainable(_MockWandbTrainableMixin, Trainable):
-        mock_finish = Mock()  # Mock call to wandb.finish()
-        mock_external_hook = Mock()  # Mock external hook method
-        mock_load_class = Mock(return_value=mock_external_hook)
-        mock_queue = Mock(get=Mock(return_value=(_QueueItem.END, None)))
-        os.environ["WANDB_PROCESS_RUN_INFO_HOOK"] = "mock_wandb_process_run_info_hook"
-
-        with patch.multiple(
-            "ray.air.callbacks.wandb.wandb", init=mock_init, finish=mock_finish
-        ), patch.multiple(
-            "ray.air.callbacks.wandb", _load_class=mock_load_class
-        ), patch.multiple(
-            "ray.air.callbacks.wandb.os", chdir=Mock()
-        ):
-            logging_process = _WandbLoggingProcess(
-                logdir="mock_logdir", queue=mock_queue, exclude=[], to_config=[]
-            )
-            logging_process.run()
-
-        mock_init.assert_called_once()
-        mock_load_class.assert_called_once_with("mock_wandb_process_run_info_hook")
-        mock_external_hook.assert_called_once_with(mock_run)
-        mock_finish.assert_called_once()
-
-        # Test `trial_log_path` of W&B run config is correctly set
-        assert mock_run.config.trial_log_path == "mock_logdir"
-
-    def testWandbLoggerConfig(self):
-        trial_config = {"par1": 4, "par2": 9.12345678}
-        trial = Trial(
-            trial_config,
-            0,
-            "trial_0",
-            "trainable",
-            PlacementGroupFactory([{"CPU": 1}]),
-            "/tmp",
-        )
+@pytest.fixture
+def wandb_env():
+    if WANDB_ENV_VAR in os.environ:
+        del os.environ[WANDB_ENV_VAR]
+    yield
+    if WANDB_ENV_VAR in os.environ:
+        del os.environ[WANDB_ENV_VAR]
 
-        if WANDB_ENV_VAR in os.environ:
-            del os.environ[WANDB_ENV_VAR]
 
-        # Read project and group name from environment variable
-        os.environ[WANDB_PROJECT_ENV_VAR] = "test_project_from_env_var"
-        os.environ[WANDB_GROUP_ENV_VAR] = "test_group_from_env_var"
-        logger = WandbTestExperimentLogger(api_key="1234")
-        logger.setup()
-        assert logger.project == "test_project_from_env_var"
-        assert logger.group == "test_group_from_env_var"
+def test_wandb_logger_config(wandb_env):
+    trial_config = {"par1": 4, "par2": 9.12345678}
+    trial = Trial(
+        trial_config,
+        0,
+        "trial_0",
+        "trainable",
+        PlacementGroupFactory([{"CPU": 1}]),
+        "/tmp",
+    )
 
-        del logger
+    if WANDB_ENV_VAR in os.environ:
         del os.environ[WANDB_ENV_VAR]
 
-        # No API key
-        with self.assertRaises(ValueError):
-            logger = WandbTestExperimentLogger(project="test_project")
-            logger.setup()
+    # Read project and group name from environment variable
+    os.environ[WANDB_PROJECT_ENV_VAR] = "test_project_from_env_var"
+    os.environ[WANDB_GROUP_ENV_VAR] = "test_group_from_env_var"
```

Comment:
Any reason we can't use `monkeypatch` here and elsewhere? I think we can also remove `wandb_env` if we do this?
```suggestion
    monkeypatch.setenv(WANDB_PROJECT_ENV_VAR, "test_project_from_env_var")
    monkeypatch.setenv(WANDB_GROUP_ENV_VAR, "test_group_from_env_var")
```

Reply:
can we make it a fixture/context manager/patch decorator instead so that the env vars are properly removed if the test throws an exception?

Reply:
updated